src/internal/recovery/recovery.go

```diff
@@ -48,15 +79,27 @@ func (id SnapshotID) String() string {
 }
 
 // createSnapshotRow creates a snapshot row containing a chunkset referencing all live data.
-func createSnapshotRow(ctx context.Context, tx *sqlx.Tx, s *fileset.Storage) (result SnapshotID, _ error) {
+func createSnapshotRow(ctx context.Context, tx *sqlx.Tx, s *fileset.Storage) (result SnapshotID, sql string, _ error) {
+	var b strings.Builder
+
+	// Create (and dump) chunkset.
 	chunksetID, err := s.CreateChunkSet(ctx, tx)
 	if err != nil {
-		return 0, errors.Wrap(err, "create chunkset")
+		return 0, "", errors.Wrap(err, "create chunkset")
 	}
-	if err := tx.GetContext(ctx, &result, `insert into recovery.snapshots (chunkset_id, pachyderm_version) values ($1, $2) returning id`, chunksetID, version.Version.String()); err != nil {
-		return 0, errors.Wrap(err, "create snapshot row")
+	b.WriteRune('\n')
+	b.WriteString("COPY storage.chunksets (id) FROM stdin;\n")
+	fmt.Fprintf(&b, "%v\n", int64(chunksetID))
```

Comment:
Shouldn't this be a log line?

Reply:
No, this is the format of the database dump.  We have to manually create it here because pg_dump can't see writes from this transaction even though its reads are in this transaction.  The statement after COPY is basically a TSV file of the table content.